doc/classes/RandomNumberGenerator.xml

```diff
@@ -17,6 +17,22 @@
 		<link title="Random number generation">$DOCS_URL/tutorials/math/random_number_generation.html</link>
 	</tutorials>
 	<methods>
+		<method name="rand_weighted">
+			<return type="int" />
+			<param index="0" name="weights" type="PackedFloat32Array" />
+			<description>
+				Returns a random index with non-uniform weights. Prints an error and returns [code]null[/code] if the array is empty.
+				[codeblocks]
+				[gdscript]
+				var array: Array[int] = [1, 2, 3, 4]
+				var weights: PackedFloat32Array = [0.1, 0.2, 0.3, 0.4]
+				var rnd = RandomNumberGenerator.new()
+				var idx : int = rnd.rand_weighted(weights)
+				print(array[idx])  # Prints either of the four numbers considering the weights.
```

Comment:
Static typing is heavily discouraged in the class reference for simplicity's sake.

I modified the example to hopefully be more explanatory:
```suggestion
				var rng = RandomNumberGenerator.new()
				
				var my_array = ["one", "two", "three, "four"]
				var weights = PackedFloat32Array([0.5, 1, 1, 2])
				
				# May prints one the four elements in `my_array`.
				# It is more likely to print "four", and less likely to print "two".
				print(my_array[rng.rand_weighted(weights)])
```

Reply:
```suggestion
				var rng = RandomNumberGenerator.new()
				
				var my_array = ["one", "two", "three, "four"]
				var weights = PackedFloat32Array([0.5, 1, 1, 2])
				
				# Prints one of the four elements in `my_array`.
				# It is more likely to print "four", and less likely to print "two".
				print(my_array[rng.rand_weighted(weights)])
```

Reply:
Good one!